### PR TITLE
docs(BitFields): remove string from BitFieldResolvable

### PR DESCRIPTION
**Pull request by papaia**
**Please describe the changes this PR makes and why it should be merged:**
Changes the documentation and internal code to use BitField flags instead of strings.
```diff
- permissions.has('SEND_MESSAGES')
+ permissions.has(Permissions.FLAGS.SEND_MESSAGES)
```
**Status**

- [x] Code changes have been tested against the Discord API, or there are no code changes
- [x] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**

- [ ] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [x] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: src/client/Client.js [hunk 1]
```diff
@@ -385,7 +385,11 @@ class Client extends BaseClient {
    * @returns {Promise<string>}
    * @example
    * client.generateInvite({
-   *   permissions: ['SEND_MESSAGES', 'MANAGE_GUILD', 'MENTION_EVERYONE'],
+   *   permissions: [
+   *    Permissions.FLAGS.SEND_MESSAGES,
+   *    Permissions.FLAGS.MANAGE_GUILD,
+   *    Permissions.FLAGS.MENTION_EVERYONE,
+   *    ],
```
Comment (kyranet) [inline src/client/Client.js:392]:
```suggestion
   *     Permissions.FLAGS.SEND_MESSAGES,
   *     Permissions.FLAGS.MANAGE_GUILD,
   *     Permissions.FLAGS.MENTION_EVERYONE,
   *   ],
```

**Review by kyranet — CHANGES_REQUESTED** [review]

**Review by kyranet — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

**Comment by iCrawl** [comment]
Can those conflicts be resolved?

Also I would still want to move forward with this PR, despite the downvotes.
But I think to eliviate all the issues people have with this, can we revert all changes made in `src/util/BitField.js`?

This way strings will still be able to be used, but not actively promoted anymore in the docs.

**Comment by papaia** [comment]
Conflicts are resolved

**Review by vladfrangu — APPROVED** [review]

File: src/util/BitField.js [hunk 2]
```diff
@@ -130,11 +130,10 @@ class BitField {
 
   /**
    * Data that can be resolved to give a bitfield. This can be:
-   * * A string (see {@link BitField.FLAGS})
-   * * A bit number
+   * * A bit number (this can be a number literal or a value taken from {@link BitField.FLAGS})
    * * An instance of BitField
    * * An Array of BitFieldResolvable
-   * @typedef {string|number|BitField|BitFieldResolvable[]} BitFieldResolvable
+   * @typedef {number|BitField|BitFieldResolvable[]} BitFieldResolvable
```
Comment (NotSugden) [inline src/util/BitField.js:136]: if its still accepted why remove it from the docs?

Comment (vladfrangu) [reply src/util/BitField.js:136]: So people use the FLAGS object more instead of strings that you can easily mistype

Comment (iCrawl) [reply src/util/BitField.js:136]: We should be fine to treat this as a hidden "feature".

Typescript will still tell you that you can use a string and the method itself won't error if you do.

Comment (NotSugden) [reply src/util/BitField.js:136]: fair enough

**Review by iCrawl — APPROVED** [review]

**Comment by papaia** [comment]
⚠️ This PR is no longer `semver: major`.

**Comment by TheFerryn** [comment]
If you remove this, I'll move to Eris 😡

**Comment by Alex-R-31** [comment]
It's just the documentation, suggesting you use Permissions.SEND_MESSAGES instead of 'SEND_MESSAGES'. No breaking changes

**Comment by adrifcastr** [comment]
> If you remove this, I'll move to Eris 😡 

Lmao, see the answer above, and also this PR has been merged already, so you'd be way to late if this was an actual breaking change

**Pull request merged**